pkg/pipeline/datasource_pipeline_templates.go

```diff
@@ -0,0 +1,94 @@
+package pipeline
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type PipelineTemplate struct {
+	ID               int    `json:"id"`
+	Name             string `json:"name"`
+	Namespace        string `json:"namespace"`
+	SyntaxVersion    string `json:"syntaxVersion"`
+	TemplateSourceId int    `json:"templateSourceId"`
+	LatestSha        string `json:"latestSha"`
+}
+
+const pipelineTemplatesUrl = "pipelines/api/v1/templates"
+
+func pipelineTemplatesDataSource() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcePipelineTemplatesRead,
+
+		Schema: map[string]*schema.Schema{
+			"templates": &schema.Schema{
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"namespace": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"syntax_version": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"template_source_id": &schema.Schema{
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"latest_sha": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourcePipelineTemplatesRead(d *schema.ResourceData, m interface{}) error {
```

Comment:
@alexhung - can we use a generic here instead of `interface{}`?

Reply:
@chb0github I don't think so since this is the func for TF resource read fun and its signature current doesn't support generics. (Also this should be a [read context func](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.17.0/helper/schema#ReadContextFunc)

Reply:
Swiched to readcontext function